tests/test_traceid.py

```diff
@@ -9,3 +12,75 @@ def test_id_format():
     assert parts[0] == '1'
     int(parts[1], 16)
     int(parts[2], 16)
+
+
+def test_id_generation_default_sampling_false():
+    segment = xray_recorder.begin_segment('segment_name', sampling=False)
+
+    # Start and end a subsegment
+    subsegment = xray_recorder.begin_subsegment('subsegment_name')
+    xray_recorder.end_subsegment()
+
+    # Close the segment
+    xray_recorder.end_segment()
+
+    assert segment.id == '0000000000000000'
+    assert segment.trace_id == '1-00000000-000000000000000000000000'
+    assert subsegment.id == '0000000000000000'
+    assert subsegment.trace_id == '1-00000000-000000000000000000000000'
+    assert subsegment.parent_id == '0000000000000000'
+
+
+def test_id_generation_default_sampling_true():
+    segment = xray_recorder.begin_segment('segment_name', sampling=True)
+
+    # Start and end a subsegment
+    subsegment = xray_recorder.begin_subsegment('subsegment_name')
+    xray_recorder.end_subsegment()
+
+    # Close the segment
+    xray_recorder.end_segment()
+
+    assert segment.id != '0000000000000000'
+    assert segment.trace_id != '1-00000000-000000000000000000000000'
+    assert subsegment.id != '0000000000000000'
+    assert subsegment.trace_id != '1-00000000-000000000000000000000000'
+    assert subsegment.parent_id != '0000000000000000'
+
+
+def test_id_generation_noop_true():
+    os.environ['AWS_XRAY_NOOP_ID'] = 'True'
+    segment = xray_recorder.begin_segment('segment_name', sampling=False)
+
+    # Start and end a subsegment
+    subsegment = xray_recorder.begin_subsegment('subsegment_name')
+    xray_recorder.end_subsegment()
+
+    # Close the segment
+    xray_recorder.end_segment()
+    os.unsetenv('AWS_XRAY_NOOP_ID')
```

Comment:
Can we unset the environment variable in a separate cleanup block? This way if one of these tests unexpectedly fails we won't have stale environment variables lying around. Example: https://github.com/aws/aws-xray-sdk-python/blob/d025f5641496efdc9dfad2b270139c224053885c/tests/test_sdk_config.py#L9-L10